### PR TITLE
Problem:(cro-1589) incorrect error message of client-rpc , when tx-qu…

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: show sync error when sync fails
also keep trying to sync until tx-query is ready again

### CODE REVIEW

File: client-rpc/src/rpc/sync_rpc.rs [hunk 1]
```diff
@@ -61,7 +62,7 @@ impl Default for SyncRequest {
     }
 }
 
-#[rpc(server)]
+#[rpc]
```
Comment (tomtau) [inline client-rpc/src/rpc/sync_rpc.rs:65]: is this annotation change necessary?

File: client-rpc/src/rpc/sync_rpc.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 use super::sync_worker::SyncWorker;
 use super::sync_worker::WorkerShared;
-use crate::to_rpc_error;
+use crate::server::to_rpc_error;
```
Comment (yihuang) [inline client-rpc/src/rpc/sync_rpc.rs:3]:
```suggestion
use crate::to_rpc_error;
```

File: client-rpc/src/rpc/sync_worker.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 use super::sync_rpc::{CBindingCallback, RunSyncProgressResult};
-use crate::rpc_error_from_string;
+use crate::server::rpc_error_from_string;
```
Comment (yihuang) [inline client-rpc/src/rpc/sync_worker.rs:2]:
```suggestion
use crate::rpc_error_from_string;
```

File: client-rpc/src/rpc/sync_rpc.rs (hunk 1 above)
Comment (yihuang) [inline client-rpc/src/rpc/sync_rpc.rs:65]:
```suggestion
#[rpc(server)]
```

**Comment by leejw51crypto** [comment]
i'll modify

File: client-rpc/src/rpc/sync_rpc.rs (hunk 1 above)
Comment (leejw51crypto) [reply client-rpc/src/rpc/sync_rpc.rs]: this is not necessary, i'll change back

**Comment by leejw51crypto** [comment]
sure, i'll do it

**Comment by tomtau** [comment]
bors try

**Pull request merged**